sonic-thermalctld/scripts/thermalctld

```diff
@@ -0,0 +1,570 @@
+#!/usr/bin/env python2
+
+"""
+    thermalctld
+    Thermal control daemon for SONiC
+"""
+
+try:
+    import os
+    import sys
+    import time
+    import signal
+    import threading
+    from datetime import datetime
+    from sonic_daemon_base import daemon_base
+    from sonic_daemon_base.daemon_base import Logger
+    from sonic_daemon_base.daemon_base import DaemonBase
+    from sonic_daemon_base.task_base import ProcessTaskBase
+except ImportError as e:
+    raise ImportError(str(e) + " - required module not found")
+
+try:
+    from swsscommon import swsscommon
+except ImportError as e:
+    from tests import mock_swsscommon as swsscommon
+
+SYSLOG_IDENTIFIER = 'thermalctld'
+NOT_AVAILABLE = 'N/A'
+logger = Logger(SYSLOG_IDENTIFIER)
+
+
+# utility functions
+
+# try get information from platform API and return a default value if caught NotImplementedError
+def try_get(callback, default=NOT_AVAILABLE):
+    """
+    Handy function to invoke the callback and catch NotImplementedError
+    :param callback: Callback to be invoked
+    :param default: Default return value if exception occur
+    :return: Default return value if exception occur else return value of the callback
+    """
+    try:
+        ret = callback()
+    except NotImplementedError:
+        ret = default
+
+    return default if ret is None else ret
```

Comment:
You made half of the change I requested. This line can now simply be changed to `return ret`

Reply:
Fixed